packages/ethers/src/client.ts

```diff
@@ -881,20 +881,28 @@ export class Web3Modal extends Web3ModalScaffold {
   private async syncProfile(address: Address) {
     const chainId = EthersStoreUtil.state.chainId
 
-    if (chainId === 1) {
-      const ensProvider = new InfuraProvider('mainnet')
-      const name = await ensProvider.lookupAddress(address)
-      const avatar = await ensProvider.getAvatar(address)
-
-      if (name) {
-        this.setProfileName(name)
-      }
-      if (avatar) {
-        this.setProfileImage(avatar)
+    try {
+      const { name, avatar } = await this.fetchIdentity({
+        address
+      })
+      this.setProfileName(name)
```

Comment:
show we use conditionals here too? e.g. some address can have `name` but not `avatar`. Not sure how this is handled internally

Reply:
The API returns `null` if there is no avatar so the conditional is redundant?